pkg/client/record/event.go

```diff
@@ -44,50 +46,94 @@ type EventSink interface {
 	Update(event *api.Event) (*api.Event, error)
 }
 
-var emptySource = api.EventSource{}
+// EventRecorder knows how to record events on behalf of an EventSource.
+type EventRecorder interface {
+	// Event constructs an event from the given information and puts it in the queue for sending.
+	// 'object' is the object this event is about. Event will make a reference-- or you may also
+	// pass a reference to the object directly.
+	// 'reason' is the reason this event is generated. 'reason' should be short and unique; it will
+	// be used to automate handling of events, so imagine people writing switch statements to
+	// handle them. You want to make that easy.
+	// 'message' is intended to be human readable.
+	//
+	// The resulting event will be created in the same namespace as the reference object.
+	Event(object runtime.Object, reason, message string)
+
+	// Eventf is just like Event, but with Sprintf for the message field.
+	Eventf(object runtime.Object, reason, messageFmt string, args ...interface{})
+}
+
+// EventBroadcaster knows how to recieve events and send them to various sources.
```

Comment:
// EventBroadcaster knows how to receive events and send them to any EventSink, watcher, or log.

?